### ISSUE TITLE
Cannot clear the value to induce defaultValue

### ISSUE THREAD

**Issue by mwisnicki** [JedWatson/react-select#4942]
`defaultValue` is used if initial `value` is undefined but once value is set to anything then new value becomes the default. I.e. subsequent undefined value will show the value that was previously set.

https://codesandbox.io/s/codesandboxer-example-forked-2j6n7?file=/example.tsx

```ts
import React, { Fragment, useState } from "react";
import Select from "react-select";

export default function Example() {
  const options = [
    { label: "zero", value: 0 },
    { label: "one", value: 1 },
    { label: "two", value: 2 }
  ];
  const [value, setValue] = useState(undefined);
  return (
    <Fragment>
      <Select
        defaultValue={options[0]}
        value={value}
        options={options}
        placeholder={"some placeholder"}
        onChange={setValue}
      />
      <button onClick={() => setValue(null)}>Set to null</button>
      <button onClick={() => setValue(undefined)}>Set to undefined</button>
      <button onClick={() => setValue([] as any)}>Set to []</button>
    </Fragment>
  );
}
```

This example starts with undefined value and correctly shows default of zero.
Change value to "one" then click "Set to undefined" and you will still see "one".
Click "Set to null" and you will get placeholder but click "Set to undefined" and it will be "one" again.

**Comment by Rall3n** [comment]
Hello @mwisnicki,

there is a clear problem with your example that contradicts basic React principles of [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html) and [controlled](https://reactjs.org/docs/forms.html#controlled-components) components.

To explain:
**Uncontrolled components** are components that manage their own state/value. They should not have a `value` prop (or in other words, `value == undefined`). If you need to set a value on those, you use the `defaultValue` prop. But this prop only works on mount / first render of the component. You usually access their state by using a ref, listening to the `onChange` handler or during form submission (using `FormData`).
**Controlled components** on the other hand are components which require you (the developing user) to manage their state by using the `value` prop. `defaultValue` will have no effect as `value` has a higher priority. To update the value you listen to the `onChange` handler and update the value-holding state.

You cannot have one input be uncontrolled and controlled at the same time.
Should you either set both `defaultValue` and `value`, or change `value` from `undefined` to any other value (or the other way around) on a default component (`<input>`, `<select>`, etc.) React will throw an error (in development mode), telling you to not do this.

We try to follow the same principle as React, but do not warn the user if they do not follow. We could consider something similar to Reacts error messages to warn the user about controlled and uncontrolled components.

-----
TL;DR:
Do not use both `defaultValue` or `value`. Decide to use either controlled or uncontrolled component, but not both.
